File: lib/app/domain/auth_state.dart
class AuthState {
  const AuthState({
    this.staffId,
    this.facilityId,
  });

  final String? staffId;
  final int? facilityId;

  AuthState copyWith({
    String? staffId,
    int? facilityId,
  }) {
    return AuthState(
      staffId: staffId ?? this.staffId,
      facilityId: facilityId ?? this.facilityId,
    );
  }

  bool canStartMonitoring() {
    return staffId != null && staffId!.isNotEmpty && facilityId != null;
  }
}

File: src/app/auth_state.js
class AuthState {
  constructor({ staffId = null, facilityId = null } = {}) {
    this.staffId = staffId;
    this.facilityId = facilityId;
  }

  canStartMonitoring() {
    return Boolean(this.staffId) && Number.isInteger(this.facilityId);
  }
}

module.exports = {
  AuthState
};
